### PR TITLE
TcpServer

### PR DESCRIPTION
**Pull request by NikitOS94**
Cretead TcpServer from HttpServer in HurmaDB

### CODE REVIEW

**Comment by NikitOS94** [comment]
Cretead TcpServer from HttpServer in HurmaDB

File: src/Main.cpp [hunk 1]
```diff
@@ -50,6 +50,7 @@ static void httpKVGetRangeHandler(const HttpRequest& req, HttpResponse& resp) {
 }
 
 static void httpKVPutHandler(const HttpRequest& req, HttpResponse& resp) {
+    bool append = false;
```
Comment (afiskon) [inline src/Main.cpp]: It seems that this variable is not used.

File: include/PersistentStorage.h [hunk 2]
```diff
@@ -14,7 +14,7 @@ class PersistentStorage : public Storage {
 
     PersistentStorage();
     ~PersistentStorage();
-    void set(const std::string& key, const std::string& value);
+    void set(const std::string& key, const std::string& value, bool* append);
```
Comment (afiskon) [inline include/PersistentStorage.h]: Doesn't look like this change has something to do with TcpServer class.

File: include/Storage.h [hunk 3]
```diff
@@ -6,7 +6,7 @@
 
 class Storage {
 public:
-    virtual void set(const std::string& key, const std::string& value) = 0;
+    virtual void set(const std::string& key, const std::string& value, bool* append) = 0;
```
Comment (afiskon) [inline include/Storage.h]: Ditto.

File: src/PersistentStorage.cpp [hunk 4]
```diff
@@ -80,6 +83,7 @@ std::string PersistentStorage::getRange(const std::string& key_from, const std::
 }
 
 void PersistentStorage::del(const std::string& key, bool* found) {
+
```
Comment (afiskon) [inline src/PersistentStorage.cpp]: I believe you forgot to run clang-format.

File: src/PersistentStorage.cpp [hunk 5]
```diff
@@ -7,6 +7,9 @@
 #include <rapidjson/writer.h>
 #include <stdexcept>
 #include <string>
+#include "rapidjson/document.h"
+#include "rapidjson/stringbuffer.h"
+#include <rapidjson/writer.h>
```
Comment (afiskon) [inline src/PersistentStorage.cpp]: These includes don't look necessary here.

File: CMakeLists.txt [hunk 6]
```diff
@@ -31,7 +31,7 @@ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Werror")
 
 add_executable(hurmadb src/Main.cpp src/Socket.cpp
   src/Storage.cpp src/PersistentStorage.cpp
-  src/HttpServer.cpp src/HttpRequest.cpp src/HttpResponse.cpp src/RegexCache.cpp)
+  src/HttpServer.cpp src/TcpServer.cpp src/PgsqlServer.cpp src/HttpRequest.cpp src/HttpResponse.cpp src/RegexCache.cpp)
```
Comment (afiskon) [inline CMakeLists.txt]: PgsqlServer.cpp is missing in this PR.

File: include/TcpServer.h [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    TcpWorker(Socket& socket, TcpHandlerListItem* handlersList);
+    void run();
+
+private:
+    Socket& _socket;
+    TcpHandlerListItem* _handlersList;
+
+    void _TCPRequest(Socket& socket);
+};
+*/
```
Comment (afiskon) [inline include/TcpServer.h]: Why there is an unused code in the PR?

File: include/TcpServer.h [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+class TcpServer {
+public:
+    TcpServer();
+    ~TcpServer();
+    void listen(const char* host, int port);
+    void accept(const std::atomic_bool& terminate_flag);
+    virtual void newThread(int accepted_socket){ throw std::runtime_error("HttpServer::accept() - malloc() call failed"+accepted_socket); };
```
Comment (afiskon) [inline include/TcpServer.h]: Well, this line definitely looks wrong.

File: src/HttpServer.cpp [hunk 9]
```diff
@@ -257,68 +253,8 @@ void HttpServer::addHandler(HttpMethod method, const char* regexpStr, HttpReques
...
-
-        /* timeout - check a terminate flag */
-        if(terminate_flag.load())
-            return;
-    }
-
-    int accepted_socket = ::accept(_listen_socket, 0, 0);
-    if(accepted_socket == -1)
-        throw std::runtime_error("HttpServer::accept() - accept() call failed");
+void HttpServer::newThread(int accepted_socket){
```
Comment (afiskon) [inline src/HttpServer.cpp]: Creating a new thread is not HTTP-specific so it should be a part of TcpServer.

**Comment by afiskon** [comment]
Closed as a duplicate of https://github.com/afiskon/hurmadb/pull/12

**Pull request closed without merging**